File: api/src/models/paste.ts
import mongoose from 'mongoose';

const pasteSchema = new mongoose.Schema({
  id: {
    type: String,
    required: true,
    unique: true
  },
  views: {
    type: Number,
    required: true,
    default: 0
  },
  encrypted: {
    type: Boolean,
    required: true,
    default: false
  },
  burn: {
    type: Boolean,
    required: true,
    default: false
  },
  value: {
    type: String,
    required: true
  }
});

export const Paste = mongoose.model('Paste', pasteSchema);

File: lib/minbin/minbin/paste.ex
defmodule Minbin.Minbin.Paste do
  use Ecto.Schema
  import Ecto.Changeset

  schema "pastes" do
    field :title, :string
    field :uuid, :integer
    field :content, :string
    field :encrypted, :boolean, default: false
    field :burn, :boolean, default: false
    field :views, :integer
    field :owner_id, :id

    timestamps(type: :utc_datetime)
  end

  @doc false
  def changeset(paste, attrs) do
    paste
    |> cast(attrs, [:uuid, :title, :content, :encrypted, :burn, :views])
    |> validate_required([:uuid, :title, :content, :encrypted, :burn, :views])
    |> unique_constraint(:uuid)
  end
end
